### PR TITLE
Collapse Magnitude with UInteger

### PR DESCRIPTION
**Pull request by rmarrowstone**
This change removes the Magnitude struct and all impl bits, in favor
of collapsing the unique functionality there with the UInteger type.

Both were an enum over u64 and BigUint, with an overlapping set of
inherent and trait impls.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/types/decimal.rs [hunk 1]
```diff
@@ -47,6 +49,12 @@ impl Decimal {
         self.coefficient.number_of_decimal_digits()
     }
 
+    /// Returns the Sign of the Decimal value.
+    /// Unscaled zero values may have either a Positive or Negative Sign.
```
Comment (zslayton) [inline src/types/decimal.rs:53]: Couple of thoughts here:
* I believe the phrase "unscaled value" means the same thing as "coefficient" -- is that correct? If so, I'd rather stick to "coefficient" for consistency with the rest of the code base.
* A coefficient of zero will _only_ have a `Negative` sign if it's truly `-0`. I think you may be referring to what will happen if we have to return the coefficient as a `u64`, but that won't impact the `Sign` value which is stored and evaluated separately. (Or have I misunderstood something?)

File: src/types/integer.rs [hunk 2]
```diff
@@ -55,9 +56,8 @@ pub trait IntAccess {
     fn as_big_int(&self) -> Option<&BigInt>;
 }
 
-/// Represents a UInt of any size. Used for reading binary integers and symbol IDs.
-///
-/// Equality tests do NOT compare across storage types; U64(1) is not the same as BigUInt(1).
+/// Represents a UInt of any size. Used for reading binary integers and symbol Ids.
+/// Use to represent the unsigned magnitude of Decimal values and fractional seconds.
```
Comment (zslayton) [inline src/types/integer.rs:60]:
```suggestion
/// Used to represent the unsigned magnitude of Decimal values and fractional seconds.
```

File: src/types/integer.rs [hunk 3]
```diff
@@ -136,6 +161,90 @@ impl From<UInteger> for Integer {
...
+
+impl_uinteger_from_small_unsigned_int_types!(u8, u16, u32, u64, usize);
+
+macro_rules! impl_uinteger_from_small_signed_int_types {
+    ($($t:ty),*) => ($(
+        impl From<$t> for UInteger {
+            fn from(value: $t) -> UInteger {
+                // Discard the sign and convert the value to a u64.
+                let magnitude: u64 = value.checked_abs()
+                        .and_then(|v| Some(v.abs() as u64))
```
Comment (zslayton) [inline src/types/integer.rs:212]: Long ago when I wrote this, Rust didn't have the [`unsigned_abs`](https://doc.rust-lang.org/std/primitive.i64.html#method.unsigned_abs) method to handle this conversion.

Comment (rmarrowstone) [reply src/types/integer.rs:210]: ack. changed to use that.

**Review by zslayton — APPROVED** [review]

**Comment by zslayton** [comment]
image

Sweet!

**Pull request merged**